### ISSUE TITLE
CSS compiler issues: comments and commas

### ISSUE THREAD

**Issue by rauchg** [zeit/styled-jsx#57]
The following doesn't compile well, unless I take out the comment

```css
.story {
  /* todo: probably a more flex-y way of doing this? */
  flex: 100;
  display: inline-block;
}
```

And based on how stylis.js was originally written, it uses `split` with `,` for selectors, where it should actually keep track that the `,` is not inside a selector for example., so this will break:

```css
.story[title=","] {
  display: inline-block;
}
```

**Comment by giuseppeg** [comment]
@rauchg fixing the comments should be trivial, do you want to preserve or strip them out?
The second case is tricker. Does it break when there are properties with multiple values separated by comma like `box-shadow: 0 0 10px black, inset 0 0 5px black` ?

**Comment by rauchg** [comment]
- I think we should strip / minify as much as possible, since this is the _compilation_ step
- I mean for selectors. Reading the code I realized [this](https://github.com/zeit/styled-jsx/blob/master/lib/style-transform.js#L197) is too naive, and @thysultan has already addressed it upstream if you need inspiration

**Comment by thysultan** [comment]
originally comments where not handled property there where edge cases that escaped, though upstream they are handled well.

**Comment by giuseppeg** [comment]
Adding one more here – the `:root` selector shouldn't be namespaced cc @nkzawa

**Comment by rauchg** [comment]
+1

**Comment by giuseppeg** [comment]
fixed by #134
